osu.Game/Online/API/APIMod.cs

```diff
@@ -52,7 +52,7 @@ public Mod ToMod(Ruleset ruleset)
             return resultMod;
         }
 
-        public bool Equals(IMod other) => Acronym == other?.Acronym;
+        public bool Equals(APIMod other) => Acronym == other?.Acronym;
```

Comment:
You can just keep the `IMod` comparison and cast in the local one to keep the implementation in one place, I believe.

Reply:
To be clear, does this mean to explicitly implement `Equals(IMod)` or the opposite?

Reply:
Much of a muchness I think. Might as well leave it public?

Reply:
Upon thinking, I believe the right way is to make `Equals(IMod)` explicit, I don't think we should be allowing stuff like `Mod.Equals(APIMod)`, right?

Reply:
That was how it was before, no? If anything that's a safer way to do it as I don't even know if there wasn't anything that was silently relying on that.

Reply:
Also I don't know if making the `IMod` implementation explicit or implicit does anything even. `EqualityComparer`s will still use it anyway via the interface specification.

Reply:
This time, if the other `IMod` is not the same type as `this`, then `false` would be returned.
https://github.com/ppy/osu/blob/5efcdbd431153a902fd510d727f9b02400afcb66/osu.Game/Rulesets/Mods/Mod.cs#L152

Reply:
`Mod` was actually already using strict type equality before this change.

Reply:
Ah, fixed.